src/settings/components/utils.ts

```diff
@@ -533,15 +552,8 @@ export const getAvailableEmailAddresses = (
 		});
 	}
 
-	// Adds all the aliases
-	if (settings.attrs.zimbraMailAlias) {
-		if (isArray(settings.attrs.zimbraMailAlias)) {
-			result.push(...(settings.attrs.zimbraMailAlias as string[]));
-		} else {
-			result.push(String(settings.attrs.zimbraMailAlias));
-		}
-	}
-
+	result.push(...getAttributeIfPresent(settings, 'zimbraMailAlias'));
+	result.push(...getAttributeIfPresent(settings, 'zimbraAllowFromAddress'));
```

Comment:
The 2 fields zimbraMailAlias and zimbraAllowFromAddress need to be added to AccountSettingsAttrs with their proper value type